src/Mod/Sketcher/App/planegcs/Constraints.h

```diff
@@ -155,6 +157,47 @@ namespace GCS
         double grad(double *) override;
     };
 
+    // Center of Gravity
+    class ConstraintCenterOfGravity : public Constraint
+    {
+    public:
+        /// Constrains that the first parameter is center of gravity of rest
+        ConstraintCenterOfGravity(const std::vector<double *>& givenpvec);
```

Comment:
We need some small documentation indicating what is inside the givenpvec and what it solves.

Please, check if you need both constructors.

Reply:
Added a documentation similar to weighted linear combination.